lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -196,6 +197,11 @@ def add_tokens(post, payment, options)
         post[:payment_method] = card_id
       end
 
+      def add_3ds_gateway_specific(post, payment, options)
```

Comment:
💬 Style Comment:

Is just style but having just one method `add_3ds` help us to be more clear and clean:

```ruby
      def add_3ds(post, payment, options)
        if options[:execute_threed] == true
          post[:payment_method_options] = { '3d_required' => true }
        elsif options[:three_d_secure].present?
          three_d_secure = options[:three_d_secure]

          post[:payment_method_options] = {}
          post[:payment_method_options]['3d_required'] = three_d_secure[:required]
          post[:payment_method_options]['3d_version'] = three_d_secure[:version]
          post[:payment_method_options][:cavv] = three_d_secure[:cavv]
          post[:payment_method_options][:eci] = three_d_secure[:eci]
          post[:payment_method_options][:xid] = three_d_secure[:xid]
          post[:payment_method_options][:ds_trans_id] = three_d_secure[:ds_transaction_id]
        end
      end

```